### PR TITLE
Peter/amm implement liquidity extrinsics methods

### PR DESCRIPTION
**Pull request by peterslany**
Resolves https://github.com/interlay/interbtc-api/issues/534

- Adding implementation of methods to add and remove liquidity.
- Adding `StableLiquidityMetaPool` class to represent meta pools.

### CODE REVIEW

File: src/parachain/amm.ts [hunk 1]
```diff
@@ -176,7 +198,7 @@ export class DefaultAMMAPI implements AMMAPI {
     private async _getStableLpTokens(): Promise<Array<StableLpToken>> {
         const stablePools = await this.api.query.zenlinkStableAmm.pools.entries();
         const stableLpTokens = stablePools.map(([key, poolData]) => {
-            const poolBase = DefaultAMMAPI.getStableBasePool(poolData.unwrap());
+            const poolBase = DefaultAMMAPI.getStablePoolInfo(poolData.unwrap());
```
Comment (bvotteler) [inline src/parachain/amm.ts:201]: I know this is pre-existing, but noticed the unwrap before checking.
Are we sure that we don't need to filter for `poolData.isSome` ?

File: src/parachain/amm.ts [hunk 2]
```diff
@@ -299,32 +321,126 @@ export class DefaultAMMAPI implements AMMAPI {
...
             // TODO: check number base for fee
-            decodeFixedPointType(poolBase.fee),
+            decodeFixedPointType(poolInfo.fee),
         ];
-
-        const lpToken = DefaultAMMAPI.getStableLpTokenFromPoolData(poolId, poolBase);
-
-        const [pooledCurrencies, apr, A, totalSupply] = await Promise.all([
+        const lpToken = DefaultAMMAPI.getStableLpTokenFromPoolData(poolId, poolInfo);
+        const [pooledCurrenciesBase, apr, A, totalSupply] = await Promise.all([
```
Comment (bvotteler) [inline src/parachain/amm.ts:350]: Maybe a good time to rename `A` to `amplificationCoefficient` or `amplCoeff` or something... not `A` :P

**Review by bvotteler — APPROVED** [review]
Can't spot anything obviously wrong. LGTM

File: src/parachain/amm.ts (hunk 2 above)
Comment (peterslany) [reply src/parachain/amm.ts]: good idea !

File: src/parachain/amm.ts (hunk 1 above)
Comment (peterslany) [reply src/parachain/amm.ts:204]: thanks, I missed it here :)

**Pull request merged**